src/Cat.java

```diff
@@ -0,0 +1,28 @@
+public class Cat extends Animals{
+
+    private static int catCount = 0;
+
+    public Cat(String animalName) {
+        super(animalName);
+        catCount++;
+    }
+
+    @Override
+    public void run(int barrier) {
+        if (barrier <= 200) {
```

Comment:
it's better to avoid `magic numbers` but use constant for them
`private final int CAT_RUN_LIMIT = 200;`